### PR TITLE
Added exercise complex-numbers for Golang track

### LINKED ISSUE #918: complex-numbers: implement exercise
[Specification](https://github.com/exercism/problem-specifications/tree/master/exercises/complex-numbers).

### PR DESCRIPTION
**Pull request by eklatzer**
Hey, added the exercise `complex-numbers` to the Golang track

Closes #918

### CODE REVIEW

**Comment by eklatzer** [comment]
Hey, for `func (n Number) Exp() Number` I had the following problem:
```
=== RUN   TestNumber_Exp
=== RUN   TestNumber_Exp/Euler's_identity/formula
    complex_numbers_test.go:123: Number({0, 3.141593}).Exp() = {-0.99999999999994 -3.464102066193935e-07}, want {-1 0}
=== RUN   TestNumber_Exp/Exponential_of_0
=== RUN   TestNumber_Exp/Exponential_of_a_purely_real_number
    complex_numbers_test.go:123: Number({1, 0}).Exp() = {2.718281828459045 0}, want {2.718282 0}
=== RUN   TestNumber_Exp/Exponential_of_a_number_with_real_and_imaginary_part
    complex_numbers_test.go:123: Number({0.693147, 3.141593}).Exp() = {-1.999999638880022 -6.928202881431823e-07}, want {-2 0}
=== RUN   TestNumber_Exp/Exponential_resulting_in_a_number_with_real_and_imaginary_part
    complex_numbers_test.go:123: Number({0.346574, 0.785398}).Exp() = {1.0000005731176131 1.0000002463225828}, want {1 1}
```
The tests failed because the calculated values were to precise. Therefore I added `func (n Number) Format() string` and added a part about it to `instructions.append.md`. I hope it is fine like that. If not I am open for suggestions how to handle it.

File: config.json [hunk 1]
```diff
@@ -2198,4 +2206,4 @@
     "used_for/cross_platform_development",
     "used_for/web_development"
   ]
-}
+}
```
Comment (andrerfcsantos) [inline config.json:2209]:
```suggestion
}

```
Missing newline.

File: exercises/practice/complex-numbers/.meta/config.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      "complex_numbers_test.go"
+    ],
+    "example": [
+      ".meta/example.go"
+    ]
+  },
+  "blurb": "Implement complex numbers.",
+  "source": "Wikipedia",
+  "source_url": "https://en.wikipedia.org/wiki/Complex_number"
+}
```
Comment (andrerfcsantos) [inline exercises/practice/complex-numbers/.meta/config.json:19]:
```suggestion
}

```
Missing newline.

File: exercises/practice/complex-numbers/complex_numbers.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package complex
+
+// Define the Number type here.
+
+func (n Number) Real() float64 {
+	panic("Please implement the Real function")
```
Comment (andrerfcsantos) [inline exercises/practice/complex-numbers/complex_numbers.go:6]:
```suggestion
	panic("Please implement the Real method")
```
Here we have methods and not really functions.

This feedback also applies to the other stubs.

File: exercises/practice/complex-numbers/complex_numbers_test.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	"math"
+	"testing"
+)
+
+func TestNumber_Real(t *testing.T) {
+	for _, tt := range realTestCases {
+		t.Run(tt.description, func(t *testing.T) {
+			n := Number{tt.in.a, tt.in.b}
+			if got := n.Real(); got != tt.want {
+				t.Errorf("Number({%v, %v}).Real() = %v, want %v", tt.in.a, tt.in.b, got, tt.want)
```
Comment (andrerfcsantos) [inline exercises/practice/complex-numbers/complex_numbers_test.go:13]:
```suggestion
				t.Errorf("Number%+v.Real() = %v, want %v", tt.in, got, tt.want)
```

This makes the output look like `Number{a:2 b:3}.Real()`, which in the context of this exercise is a valid Go expression that a student can copy/paste and use in their program, while something like `Number({2, 3}).Real()` wouldn't really work.

This feedback also applies to the other tests.

File: exercises/practice/complex-numbers/complex_numbers_test.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		})
+	}
+}
+
+func TestNumber_Add(t *testing.T) {
+	for _, tt := range addTestCases {
+		t.Run(tt.description, func(t *testing.T) {
+			n1 := Number{tt.n1.a, tt.n1.b}
+			n2 := Number{tt.n2.a, tt.n2.b}
+			if got := n1.Add(n2); got.Real() != tt.want.a || got.Imaginary() != tt.want.b {
```
Comment (andrerfcsantos) [inline exercises/practice/complex-numbers/complex_numbers_test.go:35]: For the sake of consistency, for float point comparisons, we should take the "Interest is Interesting" exercise as reference:

https://github.com/exercism/go/blob/e9ec173379d1f6a156cee8ae050e4e99797051f8/exercises/concept/interest-is-interesting/interest_is_interesting_test.go#L8-L14

(feel free to copy paste the constant + function)

File: exercises/practice/complex-numbers/complex_numbers_test.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+				t.Errorf("Number({%v, %v}).Add({%v, %v})\n got: a=%v, b=%v\nwant: a=%v, b=%v", tt.n1.a, tt.n1.b, tt.n2.a, tt.n2.b, got.Real(), got.Imaginary(), tt.want.a, tt.want.b)
```
Comment (andrerfcsantos) [inline exercises/practice/complex-numbers/complex_numbers_test.go:36]:
```suggestion
				t.Errorf("Number%+v.Add(%+v)\n got: %+v\nwant: %+v", tt.n1, tt.n2, got, tt.want)
```

Other than the suggestion of starting with `Number%+v` that was also previously suggested, here I also used the `%+v` formatting flag to simplify the code a bit, let me know what you think.

**Review by andrerfcsantos — CHANGES_REQUESTED** [review]
Thanks a lot for this!

Left some suggestions.

File: exercises/practice/complex-numbers/complex_numbers.go (hunk 3 above)
Comment (eklatzer) [reply exercises/practice/complex-numbers/complex_numbers.go:6]: For me it is a receiver function, therefore I used function, will replace it :)

**Review by eklatzer — COMMENTED** [review]
Just reviewed everything and I think I have addressed everything you requested

**Review by andrerfcsantos — APPROVED** [review]
Thanks! A new exercise is always exciting!

**Pull request merged**